docs/data/routes/docs/nextjs/troubleshooting/en.md

```diff
@@ -0,0 +1,156 @@
+
+Note that the Node/npm version used by your CI/Production environments may be different from what is used by your local environment. If a project is not pinned to a specific Node/npm version in `package.json`, it is common for deployment agents to build using either the newest version available or an environment-specific "default" version.
+
+If you need to switch Node version on your local environment for testing, 3rd party packages like [n](https://github.com/tj/n) or [nvm]() can help with this task.
+
+System Pre-requisites
+Running Node LTS version (use node -v to check)
+
+## App does not build or Home Page shows 404
+
+Use the GraphiQL interface to verify that the sitemap query returns the expected list of page routes.
+
+```
+# YOUR_PATH should be the ID of your site root (home page) in lower-case, with dashes removed
+# YOUR_LANGUAGE should be your default language, as defined in package.json
+
+query{
+  search(where: {
+    AND:[
+      {
+      name:"_path",
+      value:"YOUR_PATH"
+      },
+      {
+      name:"_language",
+      value:"YOUR_LANGUAGE"
+      },
+      {
+      name:"_hasLayout",
+      value :"true"
+      }
+    ]
+  })
+  { 
+    total
+    results {
+        url {
+          path
+        }
+      }
+  }
+}
+```
+
+If you do not see expected results, try the following steps:
+
+1. Populate Solr Schema (Sitecore Control Panel > “Populate Solr Managed Schema”)
+
+2. Rebuild indexes (Sitecore Control Panel > “Indexing Manager”)
+
+## Errors regarding SSL certificates
+
+Error examples
+> Error: unable to verify the first certificate
+>
+> Error: UNABLE_TO_VERIFY_LEAF_SIGNATURE
+
+> UnauthorizedError: invalid signature
+
+If you are working locally, make sure that you have [followed the steps](todo://add_link_to_new_page) for enabling self-signed certificates in Node. Check that you have the Node Extra Certificates path in your system variables.
+
+Node that the steps are different, depending on whether you're using a containerized or on-prem installation of Sitecore.
+
+## Errors when generating GraphQL introspection data
+
+This task is performed if the GraphQL schema changes. It depends on the `scjssconfig.json` file being present and populated.
+
+## Errors deploying a JSS app locally
+
+```
+IMPORT ERROR(S) OCCURRED!
+Exception thrown while importing JSS app
+Exception: Sitecore.Exceptions.AccessDeniedException
+Message: AddFromTemplate - Add access required
+```
+
+This occurs when the import role doesn't have the correct permissions. See the following Knowledge Base article for the solution: https://kb.sitecore.net/articles/650791
+
+## General solution checklist
+
+- Verify that all environment variables defined in the [`.env` file](https://github.com/Sitecore/jss/blob/master/samples/nextjs/.env) file are provided. On local environments, an `.env.local` file should be for this purpose.
```

Comment:
Extra "file"

Reply:
Do we suppose that user will always use .env file? User can work without env variables, he can use scjssconfig.json

Reply:
Clarified